ArrowFollowsSelected.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ArrowFollowsSelected : MonoBehaviour
+{
+    public Transform AlignTo;
+    public Canvas MyCanvas;
+    public RectTransform MyTransform;
+    public bool TrackPlanets;
+    public bool CorrectGraphSelected;
+    public Vector3 Offset;
+    private string OriginalText;
+    private bool _FirstUpdate = true;
+
+    // Use this for initialization
+    void Start()
+    {
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (_FirstUpdate)
+        {
+            ExposeCAPI();
+            _FirstUpdate = false;
+        }
+
+        if (TrackPlanets)
+        {
+            RepositionSelf();
+        }
+    }
+
+    public void ToggleTracking()
+    {
+        if (!CorrectGraphSelected)
+        {
+            TrackPlanets = !TrackPlanets;
+        }
+    }
+
+    public void RepositionSelf()
+    {
+        //Debug.Log("Aligning " + gameObject.name + " | " + MyTransform.position + " : " + Camera.main.WorldToScreenPoint(AlignTo.position));
+        MyTransform.position = Camera.main.WorldToScreenPoint(AlignTo.position) + Offset;
+    }
+
+    private void RepositionToCursor()
+    {
+        if (!CorrectGraphSelected)
+        {
+            Vector2 pos;
+            RectTransformUtility.ScreenPointToLocalPointInRectangle(MyCanvas.transform as RectTransform, Input.mousePosition, MyCanvas.worldCamera, out pos);
+            MyTransform.position = MyCanvas.transform.TransformPoint(pos);
+        }
+    }
+
+    public void BeginButtonDrag()
+    {
+
+    }
+
+    public void DuringDrag()
+    {
+        RepositionToCursor();
+    }
+
+    private void ExposeCAPI()
+    {
+        Capi.expose<bool>("UI Controller.Demo Controller." + gameObject.name + ".Enabled", () => { return gameObject.activeSelf; }, (value) => { return CapiHelper.ToggleObject(gameObject, value); });

```

Comment:
The Unity component-object model makes this more ornery - see the UIController class for an example of how that looks. Since many of the CAPI properties that our LDs want map to existing Unity variables, it's generally easier to have the objects expose their own CAPI instead of trying to setup a CAPI controller that has to pay attention to the same number of objects that other controller classes already need to be tracking. 

It's possible I suppose to create a CAPI event system that listens and pushes when changes are made, but that's a re-write of a lot of the core CAPI stuff we have been using, as well as wrangling with the event system. I wish I had time to wrestle that bear.